### PR TITLE
Enclosure support for atom feeds

### PR DESCRIPTION
**Pull request by mahemoff**
I came across a [podcast feed](http://feeds.feedburner.com/SlaughterFilm) using Atom, not the usual RSS format. So the normal iTunes parser didn't support it and I wanted a way to extract the enclosure from an Atom feed. The enclosure was included as a `link` with `rel=enclosure`, but the Atom parser strips out rel and other attributes as it just puts the links in an array of URLs. So I made this to extract the first link of type enclosure.

One caveat here is I also made the extracted properties editable using `attr_accessor`. This is because it's sometimes necessary to "massage" feed data after it's been populated from the raw feed. Normally, SaxMachine's `element(s)` declarations will automatically make the field editable; the `attr_accessor` here is emulating the same behaviour.

### CODE REVIEW

**Comment by mikeastock** [comment]
@mahemoff are you still looking to get this merged?

**Comment by mahemoff** [comment]
Yes, if possible. Been using my fork since making this update, so made those date updates subsequently; but only need to apply c8eb489 for Atom enclosures. If you can do that, I'll make a separate PR for the date formats.

File: lib/feedjira/parser/atom_entry.rb [hunk 1]
```diff
@@ -23,10 +25,26 @@ class AtomEntry
...
+        @enclosure_url ||= enclosures.first
+      end
+
+      def enclosure_type
+        @enclosure_type ||= enclosures.first
+      end
+
+      def enclosure_length
+        @enclosure_type ||= enclosures.first
+      end
```
Comment (knu) [inline lib/feedjira/parser/atom_entry.rb]: Looks to me all these methods return the same value.  Is that what you really mean?

Comment (mahemoff) [reply lib/feedjira/parser/atom_entry.rb]: No, should be `@enclosure_url ||= enclosures.first&.url` etc (or equivalent if the safe operator `&.` isn't wanted)

**Comment by doutatsu** [comment]
Damn this has been open since 2015... will this ever get merged?

**Comment by HParker** [comment]
@doutatsu you are welcome to open the PR that @mahemoff suggested with only the changes from that commit. This PR will not be merged in its current state.

**Comment by HParker** [comment]
I am going to close, but feel free to reopen if you are interested in shipping this change still. Thanks!

**Pull request closed without merging**